src/util/hotbar_source.cpp

```diff
@@ -0,0 +1,71 @@
+// Luanti
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2025 cx384
+
+#include "hotbar_source.h"
+#include "serialize.h"
+
+void HotbarSource::setHotbarItemcountLegacy(s32 count)
```

Comment:
Why `s32 count` here? If you convert to an u16, please use `rangelim` to avoid funny side-effects.